addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1708,13 +1708,21 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
         await this._super(...arguments);
         this.input = document.createElement('input');
         this.input.type = "range";
+        this.isCustomRange = false;
         let min = this.el.dataset.min && parseFloat(this.el.dataset.min) || 0;
         let max = this.el.dataset.max && parseFloat(this.el.dataset.max) || 100;
-        const step = this.el.dataset.step && parseFloat(this.el.dataset.step) || 1;
+        let step = this.el.dataset.step && parseFloat(this.el.dataset.step) || 1;
         if (min > max) {
             [min, max] = [max, min];
             this.input.classList.add('o_we_inverted_range');
         }
+        Object.keys(this.el.dataset).map(key => {
+            const dataValues = this.el.dataset[key].split(/\s*\|\s*/g);
+            if (!this.isCustomRange && dataValues.length > 1) {
+                this.isCustomRange = true;
+                [min, max, step] = [0, dataValues.length - 1, 1];
+            }
+        });
```

Comment:
Not good: You don't know each attribute you investigate in the dataset is linked to a option method. You have to use the `optionsPossibleValues` variable, initialized by the `loadMethodsData` method. I understand that this method is called after `start` but you have to find the solution to change that. Or, at worst, you can adapt the input min/max/step when calling `loadMethodsData`.

I would get rid of the `isCustomRange` attribute, it is not needed. There is nothing custom about the range, it is only "custom" depending on the method.